models/resnet.py

```diff
@@ -83,26 +85,28 @@ def __init__(self, num, num_classes, groups=1, width_per_group=64):
     self.layer2 = self._make_layer(self.block, 128, self.num_blocks[1], stride=2)
     self.layer3 = self._make_layer(self.block, 256, self.num_blocks[2], stride=2)
     self.layer4 = self._make_layer(self.block, 512, self.num_blocks[3], stride=2)
-    self.fc = nn.Linear(512 * self.block.expansion, num_classes)
+    self.fc = nn.Linear(512 * self.block.expansion, num_classes) if num_classes is not None else None
 
   def _make_layer(self, block, planes, num_blocks, stride):
     strides = [stride] + [1] * (num_blocks-1)
     layers = []
     for stride in strides:
-      layers.append(block(self.in_planes, planes, stride, self.groups, self.base_width))
+      layers.append(block(self.in_planes, planes, stride, self.stride_in_1x1, self.groups, self.base_width))
       self.in_planes = planes * block.expansion
     return layers
 
   def forward(self, x):
     out = self.bn1(self.conv1(x)).relu()
     out = out.pad2d([1,1,1,1]).max_pool2d((3,3), 2)
-    out = out.sequential(self.layer1)
-    out = out.sequential(self.layer2)
-    out = out.sequential(self.layer3)
-    out = out.sequential(self.layer4)
-    out = out.mean([2,3])
-    out = self.fc(out).log_softmax()
-    return out
+    out1 = out.sequential(self.layer1)
```

Comment:
So there's subtlety here. If the other tensors are unused (in fc mode), they should be freed ASAP. This changes the behavior.

Reply:
changed to use out everywhere, and accumulate intermediate results